File: articles/patterns-events-and-structural-explanation/r/scenario_comparison.R
# Scenario comparison using simplified structural risk logic.

article_dir <- dirname(dirname(normalizePath(sys.frame(1)$ofile)))
scenarios <- read.csv(file.path(article_dir, "data", "synthetic_scenarios.csv"))

run_scenario <- function(row, periods = 12) {
  trust <- 62
  capacity <- 55
  backlog <- 120
  structural_risk <- 52

  for (period in seq_len(periods)) {
    shock <- ifelse(row$shock_period > 0 && period >= row$shock_period, row$shock_size, 0)
    capacity <- capacity + (2 * row$capacity_multiplier) - (backlog / 150)
    backlog <- backlog + (5 * row$backlog_multiplier) + shock / 10 - (capacity / 20)
    structural_risk <- structural_risk + (backlog / 200) - (capacity / 120)
    trust <- trust + row$trust_repair_rate - (structural_risk / 120)
  }

  data.frame(
    scenario_id = row$scenario_id,
    scenario_name = row$scenario_name,
    final_trust = round(trust, 2),
    final_capacity = round(capacity, 2),
    final_backlog = round(backlog, 2),
    final_structural_risk = round(structural_risk, 2)
  )
}

results <- do.call(rbind, lapply(seq_len(nrow(scenarios)), function(i) run_scenario(scenarios[i, ])))
print(results)

File: _archive/pre-professional-script-upgrade/20260531-194442/patterns-events-and-structural-explanation/python/scenario_comparison.py
"""
Scenario comparison for event-pattern-structure dynamics.
"""

from __future__ import annotations

import csv
from pathlib import Path

SCENARIO_PATH = Path(__file__).resolve().parents[1] / "data" / "synthetic_scenarios.csv"


def load_scenarios() -> list[dict[str, str]]:
    with SCENARIO_PATH.open(newline="", encoding="utf-8") as handle:
        return list(csv.DictReader(handle))


def run_scenario(scenario: dict[str, str], periods: int = 12) -> dict[str, float | str]:
    trust = 62.0
    capacity = 55.0
    backlog = 120.0
    structural_risk = 52.0

    capacity_multiplier = float(scenario["capacity_multiplier"])
    backlog_multiplier = float(scenario["backlog_multiplier"])
    trust_repair_rate = float(scenario["trust_repair_rate"])
    shock_period = int(scenario["shock_period"])
    shock_size = float(scenario["shock_size"])

    for period in range(1, periods + 1):
        shock = shock_size if shock_period and period >= shock_period else 0.0
        capacity += (2.0 * capacity_multiplier) - (backlog / 150.0)
        backlog += (5.0 * backlog_multiplier) + shock / 10.0 - (capacity / 20.0)
        structural_risk += (backlog / 200.0) - (capacity / 120.0)
        trust += trust_repair_rate - (structural_risk / 120.0)

    return {
        "scenario_id": scenario["scenario_id"],
        "scenario_name": scenario["scenario_name"],
        "final_trust": round(trust, 2),
        "final_capacity": round(capacity, 2),
        "final_backlog": round(backlog, 2),
        "final_structural_risk": round(structural_risk, 2),
    }


def main() -> None:
    print("scenario_id,scenario_name,final_trust,final_capacity,final_backlog,final_structural_risk")
    for scenario in load_scenarios():
        row = run_scenario(scenario)
        print(
            f"{row['scenario_id']},{row['scenario_name']},{row['final_trust']},"
            f"{row['final_capacity']},{row['final_backlog']},{row['final_structural_risk']}"
        )


if __name__ == "__main__":
    main()
